xbmc/settings/dialogs/GUIDialogContentSettings.cpp

```diff
@@ -129,23 +129,18 @@ bool CGUIDialogContentSettings::OnMessage(CGUIMessage &message)
 
           dialog->Open();
           // Selected item has not changes - in case of cancel or the user selecting the same item
-          if (dialog->GetSelectedLabel() == iSelected)
+          if (!dialog->IsConfirmed() || dialog->GetSelectedItem() == iSelected)
             return true;
-          for (const auto &label : labels)
-          {
-            if (dialog->GetSelectedLabelText() == label.first)
-            {
-              m_content = static_cast<CONTENT_TYPE>(label.second);
-              break;
-            }
-          }
+
+          auto selected = labels.at(dialog->GetSelectedItem());
+          m_content = static_cast<CONTENT_TYPE>(selected.second);
```

Comment:
Should be fine. It's about loading the current selected content type. Like musicvideo, music, tvshow, movie etc.